packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -1069,6 +1078,397 @@ class _MenuItemButtonState extends State<MenuItemButton> {
+  /// gets rebuilt; for example:
+  ///
+  /// ```dart
+  /// CheckboxMenuButton(
+  ///   value: _throwShotAway,
+  ///   child: const Text('THROW'),
+  ///   onChanged: (bool? newValue) {
+  ///     setState(() {
+  ///       _throwShotAway = newValue!;
+  ///     });
+  ///   },
+  /// )
+  /// ```
+  final ValueChanged<bool?>? onChanged;
+
+  /// Called when a pointer enters or exits the button response area.
+  ///
+  /// The value passed to the callback is true if a pointer has entered button
+  /// area and false if a pointer has exited.
+  final ValueChanged<bool>? onHover;
+
+  /// Handler called when the focus changes.
+  ///
+  /// Called with true if this widget's node gains focus, and false if it loses
+  /// focus.
+  final ValueChanged<bool>? onFocusChange;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  /// The optional shortcut that selects this [MenuItemButton].
+  ///
+  /// {@macro flutter.material.menu_bar.shortcuts_note}
+  final MenuSerializableShortcut? shortcut;
+
+  /// Customizes this button's appearance.
+  ///
+  /// Non-null properties of this style override the corresponding properties in
+  /// [MenuItemButton.themeStyleOf] and [MenuItemButton.defaultStyleOf].
+  /// [MaterialStateProperty]s that resolve to non-null values will similarly
+  /// override the corresponding [MaterialStateProperty]s in
+  /// [MenuItemButton.themeStyleOf] and [MenuItemButton.defaultStyleOf].
+  ///
+  /// Null by default.
+  final ButtonStyle? style;
+
+  /// {@macro flutter.material.inkwell.statesController}
+  final MaterialStatesController? statesController;
+
+  /// {@macro flutter.material.Material.clipBehavior}
+  ///
+  /// Defaults to [Clip.none].
+  final Clip clipBehavior;
+
+  /// An optional icon to display after the [child] label.
+  final Widget? trailingIcon;
+
+  /// The widget displayed in the center of this button.
+  ///
+  /// Typically this is the button's label, using a [Text] widget.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget? child;
+
+  /// Whether the button is enabled or disabled.
+  ///
+  /// To enable a button, set its [onChanged] property to a non-null value.
+  bool get enabled => onChanged != null;
+
+  @override
+  Widget build(BuildContext context) {
+    return MenuItemButton(
+      key: key,
+      onPressed: onChanged == null ? null : () {
+        switch (value) {
+          case false:
+            onChanged!.call(true);
+            break;
+          case true:
+            onChanged!.call(tristate ? null : false);
+            break;
+          case null:
+            onChanged!.call(false);
+            break;
+        }
+      },
+      onHover: onHover,
+      onFocusChange: onFocusChange,
+      focusNode: focusNode,
+      style: style,
+      shortcut: shortcut,
+      statesController: statesController,
+      leadingIcon: ExcludeFocus(
+        child: IgnorePointer(
```

Comment:
Yes, it is there so that the checkbox doesn't get pointer events like hover, because I don't really want it being interacted with directly.  You're right that the splash radius doesn't need to be zero, I'll remove that (although it doesn't hurt anything).

Reply:
I see. Thanks!